feat: enable vmrt to boot with host virtiofs drive as root fs

yvan-sraka opened the pull request:
Allow using a host virtiofs drive _(exposed as a socket)_ as the guest root filesystem in the QEMU VM _(and successfully boot from it)!_

This tweak both the runtime flags passed to `vmrt` and the `init` programs in the initramfs. The whole setup is gated behind `VIRTIOFS` Makefile flag and a `virtiofs` Cargo feature.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/golemfactory/ya-runtime-vm?pullRequest=200) <br/>All committers have signed the CLA.

yvan-sraka committed:
feat: enable vmrt to boot with host virtiofs drive as root fs

Allow using a host virtiofs drive (exposed as a socket) as the guest
root filesystem in the QEMU VM (and successfully boot from it)!

This tweak both the runtime flags passed to `vmrt` and the `init`
programs in the initramfs. The whole setup is gated behind `VIRTIOFS`
Makefile flag and a `virtiofs` Cargo feature.
